src/language-js/print/member.js

```diff
@@ -6,16 +6,65 @@ import {
 } from "../utils/index.js";
 import { printOptionalToken } from "./misc.js";
 
+/**
+ * @typedef {import("../../common/ast-path.js").default} AstPath
+ * @typedef {import("../../document/builders.js").Doc} Doc
+ */
+
 function printMemberExpression(path, options, print) {
   const objectDoc = print("object");
-  const lookupDoc = printMemberLookup(path, options, print);
+  const lookupDoc = printMemberLookup(path, options, print, objectDoc);
+
+  return label(objectDoc.label, [objectDoc, lookupDoc]);
+}
+
+/**
+ * @param {AstPath} path
+ * @param {*} options
+ * @param {*} print
+ * @param {Doc} objectDoc
+ * @returns {Doc}
+ */
+function printMemberLookup(path, options, print, objectDoc) {
+  const doc = lookupDoc();
+  if (shouldInlineMember(path, objectDoc)) {
+    return doc;
+  }
+  return group(indent([softline, doc]));
+
+  /**
+   * @returns {Doc}
+   */
+  function lookupDoc() {
+    const property = print("property");
+    const { node } = path;
+    const optional = printOptionalToken(path);
+
+    if (!node.computed) {
+      return [optional, ".", property];
+    }
+
+    if (!node.property || isNumericLiteral(node.property)) {
+      return [optional, "[", property, "]"];
+    }
+
+    return group([optional, "[", indent([softline, property]), softline, "]"]);
+  }
+}
+
+/**
+ * @param {AstPath} path
+ * @param {*} objectDoc
```

Comment:
I want to put "Doc" type here but it requires too redundant check like:
```diff
- objectDoc.label?.memberChain
+ typeof objectDoc === "object" && "label" in objectDoc &&  typeof objectDoc.label === "object" &&  "memberChain" in objectDoc.label && objectDoc.label?.memberChain
```
Putting `@property {*} [label]` for `Doc` (it should be right), it get a bit better:
```diff
- objectDoc.label?.memberChain
+ typeof objectDoc === "object" && "label" in objectDoc &&  objectDoc.label?.memberChain
```
But I think this might still make developers to avoid typing.
Then, defining `@typedef {Partial<DocObject> & (string | DocObject | DocArray)} LooseDoc`, or just define `Doc` as it make just `objectDoc.label?.memberChain` works.
I'd like to hear member's opinion

Reply:
I mean that I will open a discussion or an issue if it is worth to discuss, or open a PR if it looks better than the current one.